src/rpcblockchain.cpp

```diff
@@ -665,6 +666,14 @@ UniValue getblockchaininfo(const UniValue& params, bool fHelp)
             "        },\n"
             "        \"reject\": { ... }      (object) progress toward rejecting pre-softfork blocks (same fields as \"enforce\")\n"
             "     }, ...\n"
+            "  ],\n"
+            "  \"upgrades\": [            (array) status of non-forward compatible upgrades\n"
+            "     {\n"
+            "        \"upgradeid\": \"xxxx\",        (string) name of upgrade\n"
```

Comment:
The use of a non-self-authenticating namespace here means either that _some_ party has authority to control use of names in that namespace or else that collisions (accidental or malicious) are possible. If there is _some_ party who has control of the use of names in that namespace, then I'm guessing that party is "whoever currently has control of https://github.com/zcash/zcash".

Reply:
A field intended as a unique identifier should use some textual encoding (hex?) of the branch ID. I would put "fluffy" names like NU0 or Overwinter into a `name` field, which should not be depended on by RPC consumers to be unique (but would inherently become so via social consensus).

Reply:
Why not make this the hex of the branch ID, and only put the name in the "info" field?

Reply:
The hex of a NU0 branch id? We don't currently have such a branch, is the plan to develop upgrades on designated branches like this, and allow consumers of the RPC to monitor said branch?

Name can go in info field, seems fine to me.